devicedefender/source/DeviceDefender.cpp

```diff
@@ -0,0 +1,202 @@
+            toMove.m_owningTask = nullptr;
+            toMove.m_lastError = AWS_ERROR_UNKNOWN;
+        }
+
+        ReportTask &ReportTask::operator=(ReportTask &&toMove) noexcept
+        {
+            if (this != &toMove)
+            {
+                this->~ReportTask();
+
+                OnTaskCancelled = std::move(toMove.OnTaskCancelled);
+                cancellationUserdata = toMove.cancellationUserdata;
+                m_allocator = toMove.m_allocator;
+                m_status = toMove.m_status;
+                m_taskConfig = std::move(toMove.m_taskConfig);
+                m_taskConfig.cancellation_userdata = this;
+                m_owningTask = toMove.m_owningTask;
+                m_lastError = toMove.m_lastError;
+
+                toMove.OnTaskCancelled = nullptr;
+                toMove.cancellationUserdata = nullptr;
+                toMove.m_allocator = nullptr;
+                toMove.m_status = ReportTaskStatus::Stopped;
+                toMove.m_taskConfig = {0};
+                toMove.m_owningTask = nullptr;
+                toMove.m_lastError = AWS_ERROR_UNKNOWN;
+            }
+
+            return *this;
+        }
+
+        ReportTaskStatus ReportTask::GetStatus() noexcept { return this->m_status; }
+
+        int ReportTask::StartTask() noexcept
+        {
+            if (this->GetStatus() == ReportTaskStatus::Ready || this->GetStatus() == ReportTaskStatus::Stopped)
+            {
+
+                this->m_owningTask = aws_iotdevice_defender_v1_report_task(this->m_allocator, &this->m_taskConfig);
+
+                if (this->m_owningTask == nullptr)
+                {
+                    this->m_lastError = aws_last_error();
+                    aws_raise_error(this->m_lastError);
+                    return AWS_OP_ERR;
+                }
+                else
+                {
+                    this->m_status = ReportTaskStatus::Running;
+                }
+            }
+            return AWS_OP_SUCCESS;
+        }
+
+        void ReportTask::StopTask() noexcept
+        {
+            if (this->GetStatus() == ReportTaskStatus::Running)
+            {
+                aws_iotdevice_defender_v1_stop_task(this->m_owningTask);
+                this->m_owningTask = nullptr;
+            }
+        }
+
+        ReportTask::~ReportTask()
+        {
+            StopTask();
+            this->m_owningTask = nullptr;
+            this->m_allocator = nullptr;
+            this->OnTaskCancelled = nullptr;
+            this->cancellationUserdata = nullptr;
+        }
+
+        ReportTaskBuilder::ReportTaskBuilder(
+            Aws::Crt::Allocator *allocator,
+            std::shared_ptr<Crt::Mqtt::MqttConnection> mqttConnection,
+            Crt::Io::EventLoopGroup &eventLoopGroup,
+            const Crt::String &thingName)
+            : m_allocator(allocator), m_mqttConnection(mqttConnection), m_thingName(thingName),
+              m_eventLoopGroup(std::move(eventLoopGroup))
```

Comment:
I'm thinking this shouldn't be a move. Why not just assign by ref?

Reply:
Good point, I'll update this and avoid using the move operation.